src/compile/axis/config.ts

```diff
@@ -1,20 +1,56 @@
-import {ScaleType} from 'vega-typings/types';
-import {array} from 'vega-util';
+import {ScaleType, SignalRef} from 'vega-typings/types';
+import {array, stringValue} from 'vega-util';
 import {AxisConfig} from '../../axis';
 import {PositionScaleChannel} from '../../channel';
 import {Config} from '../../config';
 import {isQuantitative} from '../../scale';
-import {titlecase} from '../../util';
+import {keys, titlecase} from '../../util';
+import {isSignalRef} from '../../vega.schema';
 import {getStyleConfig} from '../common';
 
-function getAxisConfigFromConfigTypes(configTypes: string[], config: Config) {
+function getAxisConfigFromConfigTypes(
+  configTypes: string[],
+  config: Config,
+  channel: 'x' | 'y',
+  orient: string | SignalRef
+) {
   // TODO: add special casing to add conditional value based on orient signal
-  return Object.assign.apply(null, [{}, ...configTypes.map(configType => config[configType])]);
+  return Object.assign.apply(null, [
+    {},
+    ...configTypes.map(configType => {
+      if (configType === 'axisOrient') {
+        const orient1 = channel === 'x' ? 'bottom' : 'left';
+        const orientConfig1 = config[channel === 'x' ? 'axisBottom' : 'axisLeft'] || {};
+        const orientConfig2 = config[channel === 'x' ? 'axisTop' : 'axisRight'] || {};
+
+        const props = new Set([...keys(orientConfig1), ...keys(orientConfig2)]);
```

Comment:
No need to make a set here if you only loop over it.

Reply:
I disagree. It's the right semantic.

Reply:
Is it because you want to remove duplicates? If you iterate over values below, an array makes as much sense as a set, imo.